libbfio-sys/build.rs

```diff
@@ -66,55 +69,189 @@ fn build_static() -> Option<PathBuf> {
         target.join("lib").display()
     );
 
-    println!("cargo:rustc-link-lib=static=bfio");
     println!(
         "cargo:rustc-link-search=native={}",
         target.join("lib").canonicalize().unwrap().to_string_lossy()
     );
 
-    Some(target)
+    target.join("include")
 }
 
-fn link_dynamic() -> Option<PathBuf> {
-    println!("cargo:rustc-link-lib=dylib=bfio");
+/// Build the lib on windows (using msbuild and libyal's vstools).
+/// Require python to be installed.
+/// This function will also add the needed folder to the `link-search` path.
+/// Return the "include" folder for the library (to be used by bindgen).
+#[cfg(target_os = "windows")]
+fn build_lib(lib_path: PathBuf, shared: bool) -> PathBuf {
+    let python_exec =
+        env::var("PYTHON_SYS_EXECUTABLE ").unwrap_or_else(|_| "python.exe".to_owned());
+
+    Command::new("powershell")
+        .arg("-File")
+        .arg("synclibs.ps1")
+        .current_dir(&lib_path)
+        .stderr(Stdio::inherit())
+        .stdout(Stdio::inherit())
+        .status()
+        .expect("synclibs failed");
+
+    Command::new("powershell")
+        .arg("-File")
+        .arg("autogen.ps1")
+        .current_dir(&lib_path)
+        .stderr(Stdio::inherit())
+        .stdout(Stdio::inherit())
+        .status()
+        .expect("autogen failed");
+
+    // The folder might not exists, but we don't care.
+    let _ = std::fs::remove_dir_all(&lib_path.join("vs2015"));
 
-    if let Ok(location) = env::var("LIBBFIO_DYNAMIC_LIBPATH") {
-        assert!(
-            PathBuf::from(&location).exists(),
-            "path passed in LIBBFIO_DYNAMIC_LIBPATH does not exist!"
-        );
-        println!("cargo:rustc-link-search=native={}", location);
+    let lib_name = lib_path.file_name().unwrap().to_string_lossy().into_owned();
 
-        return Some(location.into());
+    Command::new(&python_exec)
+        .arg("..\\..\\vstools\\scripts\\msvscpp-convert.py")
+        .arg("--extend-with-x64")
+        .arg("--output-format")
+        .arg("2015")
+        .arg(format!("msvscpp\\{}.sln", lib_name))
+        .current_dir(&lib_path)
+        .env("PYTHONPATH", "..\\..\\vstools")
+        .stderr(Stdio::inherit())
+        .stdout(Stdio::inherit())
+        .status()
+        .expect("Converting the solution failed. You might need to set PYTHON_SYS_EXECUTABLE to a working python interpreter");
```

Comment:
So this might fail on a problem with the script.
We should probably fail earlier on a bad interpreter